kubectl-cloudflow/cmd/deploy.go

```diff
@@ -180,12 +181,8 @@ func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	}
 	ownerReference = storedCR.GenerateOwnerReference()
 
-  // TODO: Temporarily disabled updating ownerReferences on secret
-  // It currently breaks Flink streamlets with config parameters.
-  // Disableling it so Cloudflow works while investigating of the problem.
-
-	//streamletNameSecretMap = deploy.UpdateSecretsWithOwnerReference(ownerReference, streamletNameSecretMap)
-	//createOrUpdateStreamletSecrets(k8sClient, namespace, streamletNameSecretMap)
+	streamletNameSecretMap = deploy.UpdateSecretsWithOwnerReference(ownerReference, streamletNameSecretMap)
+	createOrUpdateStreamletSecrets(k8sClient, namespace, streamletNameSecretMap)
```

Comment:
@skyluc Have you tested what happens if you re-configure the application? That part updates the ownerreferences also

Reply:
Hmm. No. Didn't check that.

Reply:
Tested reconfiguring a Flink streamlet. It worked fine, data was processed after restart of the pods.
Looking at the code, we only create/update the secret once. So if the problem at deploy is linked to timing, it does affect this.